### PR TITLE
fix(codebuild): missing permissions for SecretsManager environment variables

### PR DESCRIPTION
**Pull request by skinny85**
When creating a CodeBuild Project that uses environment variables from SecretsManager,
the Project fails execution with:

```
AccessDeniedException: User: arn:aws:sts::828671620168:assumed-role/role
is not authorized to perform: secretsmanager:GetSecretValue on resource:
arn:aws:secretsmanager:us-west-2:123456789012:secret:my-secret-GXyUCE
```

The solution is to automatically grant the Project's Role permissions to read all
Secrets whose names were provided as environment variables.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Review by shivlaks — APPROVED** [review]

**Comment by jogold** [comment]
The build of this PR is stuck and is blocking the mergify "queue"

@skinny85 @shivlaks 

see https://github.com/aws/aws-cdk/pull/12123/checks?check_run_id=1588463268

File: packages/@aws-cdk/aws-codebuild/test/test.project.ts [hunk 1]
```diff
@@ -889,5 +889,44 @@ export = {
 
       test.done();
     },
+
```
Comment (iliapolo) [inline packages/@aws-cdk/aws-codebuild/test/test.project.ts:892]: To trigger a build

```suggestion


```

**Comment by iliapolo** [comment]
Pushed a dummy commit to re-trigger the build since for some reason the CodeBuild job wasn't running causing mergify to be blocked

File: packages/@aws-cdk/aws-codebuild/test/test.project.ts [hunk 2]
```diff
@@ -889,5 +889,45 @@ export = {
...
+
```
Comment (iliapolo) [inline packages/@aws-cdk/aws-codebuild/test/test.project.ts:893]:
```suggestion

```

**Pull request merged**